src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -2010,8 +2010,8 @@ static void copy(MemorySegment srcSegment, ValueLayout srcElementLayout, long sr
      * @throws IllegalArgumentException if the access operation is
      * <a href="MemorySegment.html#segment-alignment">incompatible with the alignment constraint</a> in the provided layout.
      * @throws IllegalArgumentException if {@code layout.byteAlignment() > layout.byteSize()}.
-     * @throws IndexOutOfBoundsException if {@code index * byteSize()} overflows.
-     * @throws IndexOutOfBoundsException if {@code index * byteSize() > byteSize() - layout.byteSize()}.
+     * @throws IndexOutOfBoundsException if {@code index * layout.byteSize()} overflows.
+     * @throws IndexOutOfBoundsException if {@code index * layout.byteSize() > byteSize() - layout.byteSize()}.
```

Comment:
Here is the change for `setAtIndex(AddressLayout, ...)`

Reply:
Ah - true